### DISCUSSION TITLE
Automated session management

### DISCUSSION THREAD

**Discussion by dia-susca** [yaalalabs/fixyl#26] [Ideas]
It would be very useful if there was an option to have automated session management. That means that the app would manage (automatically send/reply to) messages such as: login, heartbeats, test requests, session sequence resets, resend requests, etc. 

Currently, if you accidentally send an order before a login (or any other transmission error), the sequence numbers become invalid and then you have to reset the session sequence numbers using special messages, and correctly answer to resend requests. That's a hassle, if someone is solely interested in sending business messages.

So, maybe in the future you can add this feature, to have the option to not worry about managing session/transport information. 
Maybe even an extra "reset sequence number" button somewhere, to be able to do that during an open session.

**Comment by DhananjayaR** [comment]
Hi @dia-susca, thanks for your valuable suggestion. I'll add this as a feature in a future release for sure and give an update on this thread. Thanks again for showing interest in the project ✌️

**Reply by DhananjayaR** [reply]
Hi @dia-susca, the feature you've requested is implemented in https://github.com/yaalalabs/fixyl/releases/tag/v1.0.9. It also has the capability to automatically send a logon message as soon as the connection is established. So that you can avoid mistakenly sending data before putting a logon message.
